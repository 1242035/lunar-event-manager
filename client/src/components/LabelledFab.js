import React from 'react';

/**
 * The smaller fab that contains an action.
 * @param{undefined}
 * @return{React.Component}
 */
class LabelledFab extends React.Component {
  handleKeyDown = () => {}

  showModal = () => {
    // Calling callbacks depending on which prop is passed to LabelledFab.
    if (this.props.showAddCenterModal) {
      this.props.showAddCenterModal();
    } else if (this.props.showAddEventModal) {
      this.props.showAddEventModal();
    }
  }

  render() {
    const {
      position, label, icon,
    } = this.props;

    // Hide first two fab labels if user not admin.
    const hide = position > 1 && !(localStorage.getItem('user.admin') === 'true') ? 'hide' : '';

    return (
      <div id={`fab-${position}`} className={`io-labelled-fab io-${position} ${hide}`} onClick={this.showModal}>
        <span className="label">{label}</span>
        <button><i className={`fa fa-${icon}`} /></button>
      </div>
    );
  }
}

export default LabelledFab;
