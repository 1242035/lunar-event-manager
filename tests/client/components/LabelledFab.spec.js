import React from 'react';
import { shallow } from 'enzyme';
import LabelledFab from '../../../client/src/components/LabelledFab';

describe('LabelledFab', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<LabelledFab />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render component', () => {
    wrapper = shallow(<LabelledFab />);
    expect(wrapper.length).toEqual(1);
  });
});
