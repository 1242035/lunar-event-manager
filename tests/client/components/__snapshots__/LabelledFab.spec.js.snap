// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelledFab Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabelledFab />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="label"
        />,
        <button>
          <i
            className="fa fa-undefined"
          />
        </button>,
      ],
      "className": "io-labelled-fab io-undefined ",
      "id": "fab-undefined",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "label",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-undefined"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="label"
          />,
          <button>
            <i
              className="fa fa-undefined"
            />
          </button>,
        ],
        "className": "io-labelled-fab io-undefined ",
        "id": "fab-undefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "label",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-undefined"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
